server/encoder/video_encoder_nvenc.h

```diff
@@ -62,7 +54,7 @@ class video_encoder_nvenc : public video_encoder
 	int bitrate;
 
 public:
-	video_encoder_nvenc(wivrn_vk_bundle & vk, encoder_settings & settings, float fps, uint8_t stream_idx);
+	video_encoder_nvenc(std::shared_ptr<video_encoder_nvenc_shared_state> state, wivrn_vk_bundle & vk, encoder_settings & settings, float fps, uint8_t stream_idx);
```

Comment:
It is possible to hide the shared state object inside the constructor.

You can use the same pattern used in `video_encoder::sender`:
* Add a `static std::shared_ptr<video_encoder_nvenc_shared_state> get()` method in `video_encoder_nvenc_shared_state`. You can basically copy-paste the the method and just change the types.
* In the encoder constructor, assign the pointer from the get method.